### PR TITLE
Fixing chiventure segfault when trying to load wdl file

### LINKED ISSUE #1030: WDL path seg fault
Currently when the user inputs an incorrect wdl file path chiventure causes a segfault, instead of just saying something along the lines of "Invalid file path!"

### PR DESCRIPTION
**Pull request by samdonio**
Previously chiventure would have a seg fault when the user tried using the load_wdl command at all. This is because the cli didn't correctly check if the path was valid, and also because the wdl module doesn't currently load wdl files correctly. Now, the cli returns that the path is invalid and that the wdl loading isn't complete yet and doesn't seg fault anymore.

Closes #1030

### CODE REVIEW

**Comment by samdonio** [comment]
After more changes to the way that load_wdl_operation works, the LOAD_WDL command now checks whether the file exists, then checks if the file has the ending of .wdl, and if either of these are not true it tells the user that the path/file is invalid. Also, the function now correctly loads a chiventure game if the path is correct, meaning that the user can now use LOAD_WDL to load wdl files from inside the game whenever they want.

File: src/cli/src/operations.c [hunk 1]
```diff
@@ -53,11 +54,56 @@ bool validate_filename(char *filename)
     }
 }
 
+bool validate_wdl_filename(char *filename)
```
Comment (ecoble) [inline src/cli/src/operations.c:57]: I think it is safe to remove the previous validate_filename function, I don't think we use .dat files anymore for Chiventure.

File: src/cli/src/operations.c [hunk 2]
```diff
@@ -53,11 +54,56 @@ bool validate_filename(char *filename)
...
+{
+    int len = strlen(filename);
+    int min_filename_length = 4;
+    if(len < min_filename_length)
+    {
+        return false;
+    }
+    const char *ending = &filename[len - 4];
```
Comment (ecoble) [inline src/cli/src/operations.c:65]: 4 is a bit of a magic number here, can we create another variable, maybe something like "file_extension_length"

**Review by ecoble — CHANGES_REQUESTED** [review]

File: src/cli/src/operations.c (hunk 2 above)
Comment (samdonio) [reply src/cli/src/operations.c]: file_extension_length int has been made

File: src/cli/src/operations.c (hunk 1 above)
Comment (samdonio) [reply src/cli/src/operations.c:38]: This function has been removed from the operations.c and operations.h file

File: src/cli/src/operations.c [hunk 3]
```diff
@@ -33,31 +34,58 @@ char *hist_operation(char *tokens[TOKEN_LIST_SIZE], chiventure_ctx_t *ctx)
...
 }
 
-bool validate_filename(char *filename)
+
+bool validate_wdl_filename(char *filename)
 {
     int len = strlen(filename);
     int min_filename_length = 4;
+    int file_extension_length = 4;
     if(len < min_filename_length)
```
Comment (ecoble) [inline src/cli/src/operations.c:43]: Spacing after "if"

File: src/cli/src/operations.c [hunk 4]
```diff
@@ -33,31 +34,58 @@ char *hist_operation(char *tokens[TOKEN_LIST_SIZE], chiventure_ctx_t *ctx)
...
-    if(cmp == 0)
+    const char *ending = &filename[len - file_extension_length];
+    int cmp = strcmp(ending, ".wdl");
+    if (cmp == 0)
     {
         return true;
     }
-    else
-    {
+    else{
```
Comment (ecoble) [inline src/cli/src/operations.c:53]: Looks like the rest of this file is using allman style brackets, lets stick with that.

File: src/cli/src/operations.c [hunk 5]
```diff
@@ -33,31 +34,58 @@ char *hist_operation(char *tokens[TOKEN_LIST_SIZE], chiventure_ctx_t *ctx)
...
+    {
+        ch = tolower(tokens[1][i]);
+        tokens[1][i] = ch;
+        i++;
+    }
+
+
+    valid_path = access(tokens[1], F_OK);
+
+    if(valid_path == -1) //Triggers if file does not exist
```
Comment (ecoble) [inline src/cli/src/operations.c:82]: Spacing

File: src/cli/src/operations.c [hunk 6]
```diff
@@ -66,7 +94,7 @@ char *load_wdl_operation(char *tokens[TOKEN_LIST_SIZE], chiventure_ctx_t *ctx)
 
     if(game == NULL)
```
Comment (ecoble) [inline src/cli/src/operations.c:95]: Spacing

**Review by ecoble — CHANGES_REQUESTED** [review]
Looks great! Just a couple style fixes, you can probably just fix them with astyle.

**Review by dchen0258 — APPROVED** [review]
Looks good! Very neat solution.

**Comment by ecoble** [comment]
Issue Score: ✔️+

Comments:
Great job, thanks for fixing this! Be sure to put closing statements on your PRs as well, just as you would for issues.

**Pull request merged**